x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/ManageOwnApiKeyClusterPrivilege.java

```diff
@@ -92,6 +93,8 @@ protected boolean extendedCheck(String action, TransportRequest request, Authent
                 }
             } else if (request instanceof final QueryApiKeyRequest queryApiKeyRequest) {
                 return queryApiKeyRequest.isFilterForCurrentUser();
+            } else if (request instanceof GrantApiKeyRequest) {
+                return false;
             }
             throw new IllegalArgumentException(
                 "manage own api key privilege only supports API key requests (not " + request.getClass().getName() + ")"
```

Comment:
I think we should add an assertion before the throw because it is certainly a programming error for requests not related to API keys to reach here. It also helps to catch similar issue eariler in the future.